models/vehicle.rb

```diff
@@ -118,8 +118,8 @@ def self.create(hash)
     end
 
     def need_matrix_time?
-      cost_time_multiplier != 0 || cost_late_multiplier && cost_late_multiplier != 0 || cost_setup_time_multiplier != 0 ||
-        !rests.empty? || maximum_ride_time || duration || overall_duration
+      cost_time_multiplier.positive? || (timewindow.end && (cost_late_multiplier.nil? || cost_late_multiplier&.zero?)) || cost_late_multiplier&.positive? ||
```

Comment:
why `(timewindow.end && (cost_late_multiplier.nil? || cost_late_multiplier&.zero?))` instead of just `timewindow.end` like follows because if `cost_late_multiplier&.positive?` doesn't hold the extra bit `(cost_late_multiplier.nil? || cost_late_multiplier&.zero?)` always holds

```suggestion
      cost_time_multiplier.positive? || cost_late_multiplier&.positive? || timewindow.end
```

Reply:
I still got the exception case for vroom in mind. But, Indeed, it shouldn't impact the need of time dimension.